app/models/current_user_data.rb

```diff
@@ -0,0 +1,43 @@
+CurrentUserData = Struct.new(:user, :restricted) do
+  def data
+    serializer.new(user, scope: serializer_scope, root: :current_user).as_json.tap do |json|
+      json[:current_user].except!(:group_ids, :membership_ids)
+    end
+  end
+
+  private
+
+  def serializer
+    if restricted
+      Restricted::UserSerializer
+    else
+      Full::UserSerializer
+    end
+  end
+
+  def serializer_scope
+    { memberships: memberships }.tap do |hash|
+      hash.merge!(
```

Comment:
Oh, yes, it's a bit odd, eh. When the DiscussionSerializer is doing it's thang, it checks the scope to see if there is a `:reader_cache`, and pulls readers from that, so that we can do one reader query for _all_ discussions being serialized, rather than one query for _each_ discussion being serialized (this is the same reasoning behind serializing out all memberships, or all notifications here, rather than passing them through `has_many`).

This scope is also shared by every model that is serialized, so if a user is serializing out a bunch of memberships, the UserSerializer and MembershipSerializer are looking at the same scope.

So, we're not actually serializing out discussion readers separately, just loading them together so that the DiscussionSerializer can pick through the ones that it wants without making a separate query.

(We need to do this because there's no inbox query anymore, so this is where we get all unread threads for a user now, and in order for the client to think they're unread it needs the reader data fused onto the discussion)